tests/test_workers/test_scm.py

```diff
@@ -294,19 +294,21 @@ def test_create_and_verify_archive(fake_repo, caplog):
     assert f"Verifying the archive at {git_obj.sources_dir.archive_path}" in caplog.text
 
 
-def test_clone_and_verify_archive(fake_repo, caplog):
+@pytest.mark.parametrize("gitsubmodule", [True, False])
+def test_clone_and_verify_archive(fake_repo, caplog, gitsubmodule):
     repo_dir, _ = fake_repo
     git_obj = scm.Git(f"file://{repo_dir}", "master")
-    git_obj.clone_and_archive()
+    git_obj.clone_and_archive(gitsubmodule)
     assert f"Verifying the archive at {git_obj.sources_dir.archive_path}" in caplog.text
 
 
-def test_update_and_verify_archive(fake_repo, caplog):
+@pytest.mark.parametrize("gitsubmodule", [True, False])
+def test_update_and_verify_archive(fake_repo, caplog, gitsubmodule):
     repo_dir, _ = fake_repo
     git_obj = scm.Git(f"file://{repo_dir}", "master")
-    git_obj.clone_and_archive()
+    git_obj.clone_and_archive(gitsubmodule)
     caplog.clear()
-    git_obj.update_and_archive(git_obj.sources_dir.archive_path)
+    git_obj.update_and_archive(git_obj.sources_dir.archive_path, gitsubmodule)
```

Comment:
Should we test if update_git_submodules was called here?

Reply:
We could technically do but I don't see a value in doing so since that is already being tested at https://github.com/release-engineering/cachito/blob/master/tests/test_workers/test_scm.py#L273

Adding it here won't change the coverage and so, I'd rather stick to the title of the unit test to make tests easy to read.